packages/framework/ai-collab/src/aiCollabApi.ts

```diff
@@ -0,0 +1,145 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import type { TreeNode, TreeView, ImplicitFieldSchema } from "@fluidframework/tree";
+// eslint-disable-next-line import/no-named-as-default
+import type OpenAI from "openai";
+
+/**
+ * OpenAI client options for the {@link AiCollabOptions} interface.
+ *
+ * @alpha
+ */
+export interface OpenAiClientOptions {
+	/**
+	 * The {@link OpenAI} client to use for the AI collaboration.
+	 */
+	client: OpenAI;
+	/**
+	 * The name of the target OpenAI model to use for the AI collaboration.
+	 */
+	modelName?: string;
+}
+
+/**
+ * Options for the AI collaboration.
+ *
+ * @alpha
+ */
+export interface AiCollabOptions<TSchema extends ImplicitFieldSchema> {
+	openAI: OpenAiClientOptions;
```

Comment:
Unless there is a reason not to, we should make all of the interface properties `readonly`.

Reply:
Same with the other interfaces we export - probably worth an audit pass.

Reply:
All interface members, except TokenUsage's members (because they need to be updated with each llm call), have been marked readonly as of commit e835b1c037abfbab72df0c82e66781f54a4bdfe9